test/odbc/psqlodbc/test/image.cpp

```diff
@@ -0,0 +1,145 @@
+    test_teardown.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+};
+
+TEST_F(PSQL_DataTypes_Image, Table_Creation) {
+  const vector<int> LENGTH_EXPECTED = {4, 255};
+  const vector<int> PRECISION_EXPECTED = {0, 0};
+  const vector<int> SCALE_EXPECTED = {0, 0};
+  const vector<string> NAME_EXPECTED = {"int4", "unknown"};
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  testCommonColumnAttributes(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS.size(), COL1_NAME, LENGTH_EXPECTED, 
+    PRECISION_EXPECTED, SCALE_EXPECTED, NAME_EXPECTED);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Image, Insertion_Success) {
+  const vector<string> INSERTED_VALUES = {
+    "0x" + INSERTED_IMAGE_HEX_STR
+  };
+
+  const int NUM_OF_DATA = INSERTED_VALUES.size();
+
+  const string INSERT_STRING = "(0, DECODE('" + INSERTED_IMAGE_HEX_STR + "', 'hex'))";
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  insertValuesInTable(ServerType::PSQL, TABLE_NAME, INSERT_STRING, NUM_OF_DATA);
+  verifyValuesInObject(ServerType::PSQL, TABLE_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Image, Update_Success) {
+  const vector<string> INSERTED_VALUES = {
+    "0x" + INSERTED_IMAGE_HEX_STR
+  };
+  const vector<string> UPDATED_VALUES = {
+    "DECODE('" + UPDATED_IMAGE_HEX_STR + "', 'hex')"
+  };
+  const vector<string> EXPECTED_UPDATED_VALUES = {
+    "0x" + UPDATED_IMAGE_HEX_STR 
+  };
+
+  const int NUM_OF_DATA = INSERTED_VALUES.size();
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  
+  const string INSERTED_STRING = "(0, DECODE('" + INSERTED_IMAGE_HEX_STR + "', 'hex'))";
+
+  insertValuesInTable(ServerType::PSQL, TABLE_NAME, INSERTED_STRING, NUM_OF_DATA);
+  verifyValuesInObject(ServerType::PSQL, TABLE_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+  testUpdateSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, COL2_NAME, UPDATED_VALUES, EXPECTED_UPDATED_VALUES, false, true);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Image, View_creation) {
+  const vector<string> INSERTED_VALUES = {
+    "0x" + INSERTED_IMAGE_HEX_STR
+  };
+
+  const int NUM_OF_DATA = INSERTED_VALUES.size();
+
+  const string INSERT_STRING = "(0, DECODE('" + INSERTED_IMAGE_HEX_STR + "', 'hex'))";
+  const string VIEW_QUERY = "SELECT * FROM " + TABLE_NAME;
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  insertValuesInTable(ServerType::PSQL, TABLE_NAME, INSERT_STRING, NUM_OF_DATA);
+  verifyValuesInObject(ServerType::PSQL, TABLE_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+
+  createView(ServerType::PSQL, VIEW_NAME, VIEW_QUERY);
+  verifyValuesInObject(ServerType::PSQL, VIEW_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+
+  dropObject(ServerType::PSQL, "VIEW", VIEW_NAME);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
```

Comment:
Can we have a set of tests that uses operators like DATALENGTH and Substring? According to this: https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/t-sql/data-types/ntext-text-and-image-transact-sql.md, there's a list of functions and statements that could be used for the image datatype.  If we choose to pursue this, please first make sure that they are supported in BBF. If it isn't we could probably skip them.